lib/Doctrine/Persistence/Mapping/AbstractClassMetadataFactory.php

```diff
@@ -232,6 +238,7 @@ public function getMetadataFor($className)
             if ($this->cache) {
                 $cached = $this->cache->getItem($this->getCacheKey($realClassName))->get();
                 if ($cached instanceof ClassMetadata) {
+                    /** @psalm-var CMTemplate $cached */
```

Comment:
This feels like cheating, but without trying to wrap the cache interface, I'm not sure we can do better